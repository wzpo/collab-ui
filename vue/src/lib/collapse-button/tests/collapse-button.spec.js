import { mount, shallowMount } from '@vue/test-utils';
import CollapseButton from '../index.vue';
import Button from '../../button/index.vue';
import Icon from '../../icon/index.vue';
import Vue from 'vue';

Vue.component(Button.name, Button);
Vue.component(Icon.name, Icon);

describe('CollapseButton', () => {
  it('should match snapshot', () => {
    const container = mount(CollapseButton);
    expect(container.html()).toMatchSnapshot();
  });

  it('should call onClick callback when the button is clicked', () => {
    const onClick = jest.fn();
    const container = mount(CollapseButton, {
      listeners: {
        click: onClick
      }
    });
    container.trigger('click');

    expect(onClick).toHaveBeenCalled();
  });

  it('when collapse is true it should display right icon and label', () => {
    const container = mount(CollapseButton, {
      propsData: {
        collapse: true
      }
    });
    expect(container.find(Icon).props().name).toEqual('panel-control-right_12');
    expect(container.find(Button).props().ariaLabel).toEqual('expand');
  });

  it('when collapse is false it should display right icon and label', () => {
    const container = mount(CollapseButton, {
      propsData: {
        collapse: false
      }
    });
    expect(container.find(Icon).props().name).toEqual('panel-control-left_12');
    expect(container.find(Button).props().ariaLabel).toEqual('collapse');
  });

  describe('should apply correct alignment classes', () => {

    it('when alignment is left', () => {
      const container = shallowMount(CollapseButton, {
        propsData: {
          alignment: 'left'
        }
      });
      expect(container.find('.md-collapse-button').classes('md-collapse-button--left')).toEqual(true);
    });

    it('when alignment is right', () => {
      const container = shallowMount(CollapseButton, {
        propsData: {
          alignment: 'right'
        }
      });
      expect(container.find('.md-collapse-button').classes('md-collapse-button--right')).toEqual(true);
    });

    it('when alignment is top', () => {
      const container = shallowMount(CollapseButton, {
        propsData: {
          alignment: 'top'
        }
      });
      expect(container.find('.md-collapse-button').classes('md-collapse-button--top')).toEqual(true);
    });

    it('when alignment is down', () => {
      const container = shallowMount(CollapseButton, {
        propsData: {
          alignment: 'bottom'
        }
      });
      expect(container.find('.md-collapse-button').classes('md-collapse-button--bottom')).toEqual(true);
    });
  });
});

