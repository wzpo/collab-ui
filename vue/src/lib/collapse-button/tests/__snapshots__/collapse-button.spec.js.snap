// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseButton should match snapshot 1`] = `<button aria-label="expand" alt="expand" type="button" tabindex="0" role="button" class="md-collapse-button md-button md-button--36 md-collapse-button--left"><span class="md-button__children" style="opacity: 1;"><i class="md-icon icon icon-panel-control-right_12" style="font-size: 12px;"></i></span></button>`;
